### PR TITLE
Align changes from helm private repo with public repo

### PR DESCRIPTION
**Pull request by aniruddhg-nx**
Jira link: https://jira.nutanix.com/browse/ENG-562338

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -7,12 +7,14 @@ jobs:
...
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
       - name: Set up chart-testing
-        uses: helm/chart-testing-action@v2.2.1
+        uses: helm/chart-testing-action@v2.4.0
+        with:
+          version: v3.6.0
```
Comment (subodh01) [inline .github/workflows/test.yaml]: line 16-17 is not present in nutanix/helm master branch.

File: charts/nutanix-csi-snapshot/Chart.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
 apiVersion: v2
 type: application
-version: 1.0.0
+version: 6.0.2
```
Comment (subodh01) [inline charts/nutanix-csi-snapshot/Chart.yaml:3]: Please rebase all files in nutanix-csi-snapshot again with nutanix/helm master branch

File: charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml [hunk 3]
```diff
@@ -2,27 +2,44 @@
...
+            weight: 100
+      serviceAccount: nutanix-csi-controller
       hostNetwork: true
       containers:
         - name: csi-provisioner
+          {{- if semverCompare ">=1.20-0"  .Capabilities.KubeVersion.Version }}
           image: {{ .Values.sidecars.provisioner.image }}
+          {{- else }}
+          image: {{ .Values.sidecars.provisioner.imageLegacy }}
+          {{- end }}
```
Comment (subodh01) [inline charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml:42]: change line 54 to:
            - --v=2

File: charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml [hunk 4]
```diff
@@ -96,7 +114,7 @@ spec:
           imagePullPolicy: {{ .Values.imagePullPolicy }}
           args:
           - --csi-address=$(ADDRESS)
-          - --leader-election=false
+          - --leader-election=true
```
Comment (subodh01) [inline charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml:117]: Add resources for csi-snapshotter like in master branch
          resources:
            requests:
              cpu: 5m
              memory: 30Mi

File: charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml [hunk 5]
```diff
@@ -74,10 +92,10 @@ spec:
           image: {{ .Values.sidecars.resizer.image }}
           imagePullPolicy: {{ .Values.imagePullPolicy }}
           args:
-            - --v=5
+            - --v=2
```
Comment (subodh01) [inline charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml:95]: Remove the limits on all containers, just keep the requests as in master branch.

File: charts/nutanix-csi-storage/templates/ntnx-csi-node-ds.yaml [hunk 6]
```diff
@@ -5,25 +5,29 @@
...
   namespace: {{ .Release.Namespace }}
 spec:
   selector:
     matchLabels:
-      app: csi-node-ntnx-plugin
+      app: nutanix-csi-node
+  updateStrategy:
+    type: "RollingUpdate"
+    rollingUpdate:
+      maxUnavailable: 1
```
Comment (subodh01) [inline charts/nutanix-csi-storage/templates/ntnx-csi-node-ds.yaml]: maxUnavailable: 10%

File: charts/nutanix-csi-storage/templates/service-prometheus-csi.yaml [hunk 7]
```diff
@@ -6,14 +6,14 @@
...
+  name: nutanix-csi-metrics
   namespace: {{ .Release.Namespace }}
   labels:
-    app: csi-provisioner-ntnx-plugin
+    app: nutanix-csi-metrics
 spec:
   type: ClusterIP
   selector:
-    app: csi-provisioner-ntnx-plugin
+    app: nutanix-csi-controller
```
Comment (subodh01) [inline charts/nutanix-csi-storage/templates/service-prometheus-csi.yaml:16]: Add monitoring for other services also like in : https://github.com/subodh01/helm/blob/master/charts/nutanix-csi-storage/templates/service-prometheus-csi.yaml

**Review by subodh01 — COMMENTED** [review]
Thanks for working on this. It looks mostly good but you need to rebase it again and also get changes from nutanix/helm.

File: charts/nutanix-csi-storage/templates/service-prometheus-csi.yaml (hunk 7 above)
Comment (aniruddhg-nx) [reply charts/nutanix-csi-storage/templates/service-prometheus-csi.yaml:16]: done

File: charts/nutanix-csi-storage/templates/ntnx-csi-node-ds.yaml (hunk 6 above)
Comment (aniruddhg-nx) [reply charts/nutanix-csi-storage/templates/ntnx-csi-node-ds.yaml]: done

File: charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml (hunk 5 above)
Comment (aniruddhg-nx) [reply charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml:95]: - --v=2 is there on master as well. Can you pls elaborate what limit we want to remove

File: charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml (hunk 4 above)
Comment (aniruddhg-nx) [reply charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml:117]: done

File: charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml (hunk 3 above)
Comment (aniruddhg-nx) [reply charts/nutanix-csi-storage/templates/ntnx-csi-controller-deployment.yaml:42]: done

File: .github/workflows/test.yaml (hunk 1 above)
Comment (aniruddhg-nx) [reply .github/workflows/test.yaml]: done

**Review by subodh01 — APPROVED** [review]

**Pull request merged**